identity/src/main/java/com/stripe/android/identity/viewmodel/CameraViewModel.kt

```diff
@@ -0,0 +1,85 @@
+package com.stripe.android.identity.viewmodel
+
+import android.util.Log
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.stripe.android.camera.framework.AggregateResultListener
+import com.stripe.android.camera.framework.AnalyzerLoopErrorListener
+import com.stripe.android.identity.IdentityVerificationSheetContract
+import com.stripe.android.identity.IdentityViewModel
+import com.stripe.android.identity.camera.IDDetectorAggregator
+import com.stripe.android.identity.camera.IdentityScanFlow
+import com.stripe.android.identity.states.ScanState
+
+/**
+ * ViewModel hosted by all fragments that need to access live camera feed and callbacks.
+ */
+internal class CameraViewModel :
+    ViewModel(),
+    AnalyzerLoopErrorListener,
+    AggregateResultListener<IDDetectorAggregator.InterimResult, IDDetectorAggregator.FinalResult> {
+
+    // liveData for results, subscribed from fragments
+    internal val interimResults = MutableLiveData<IDDetectorAggregator.InterimResult>()
+    internal val finalResult = MutableLiveData<IDDetectorAggregator.FinalResult>()
+    internal val reset = MutableLiveData<Unit>()
+
+    lateinit var identityScanFlow: IdentityScanFlow
+
+    /**
+     * Initialize [identityScanFlow] with the target scanType.
+     *
+     * TODO(ccen): Extract scanType from [IdentityScanFlow]'s constructor, initialize the scan flow
```

Comment:
`initializeScanFlow` will be removed in a follow up PR, instead `IdentityScanFlow` will be created upon initialization. scanType can be updated while IdentityScanFlow is already started.

This is required for Identity's ID and driver license scanning - we want to use the same `IdentityScanFlow` to capture two images(front and back) in the same fragment.